cmd/apply/cmdapply.go

```diff
@@ -137,7 +138,10 @@ func (r *ApplyRunner) RunE(cmd *cobra.Command, args []string) error {
 
 	// Run the applier. It will return a channel where we can receive updates
 	// to keep track of progress and any issues.
-	a, err := apply.NewApplier(r.factory, invClient)
+	a, err := apply.NewApplierBuilder().
+		WithFactory(r.factory).
```

Comment:
Should this optional parameter be required?

Reply:
It is not required if all other things, that are derived from the factory, are provided explicitly. This approach allows user to customize what applier uses without creating their own `Factory` implementation (to inject desired objects).

Reply:
Yeah, allowing users to provide the different dependencies directly instead of through the `Factory` is quite useful.